### PR TITLE
libselinux: Fix cross-compilation

### PR DESCRIPTION
**Pull request by jwillikers**
Apply patches for 3.6 to fix detection of available functions when cross-compiling. Use the `PKG_CONFIG_LIBDIR` to point pkg-config to the correct directory containing Conan's generated pkg-config files. Without this, the `pcre2` from the system will be picked up before the one provided by Conan. When cross-compiling, this causes system include directories to poison the build.

[libselinux-3.6-cross-fixed-log.txt](https://github.com/conan-io/conan-center-index/files/14743730/libselinux-3.6-cross-fixed-log.txt)
[libselinux-3.6-cross-broken-log.txt](https://github.com/conan-io/conan-center-index/files/14743733/libselinux-3.6-cross-broken-log.txt)

### CODE REVIEW

**Review by valgur — APPROVED** [review]

File: recipes/libselinux/all/conanfile.py [hunk 1]
```diff
@@ -87,7 +87,9 @@ def generate(self):
         sepol_lib_folder = os.path.join(self._sepol_source_folder, "src")
         tc.extra_ldflags.append(f"-L{sepol_lib_folder}")
         tc.make_args.append("USE_PCRE2=y")
-        tc.generate()
+        env = tc.environment()
+        env.append_path("PKG_CONFIG_LIBDIR", self.generators_folder)
+        tc.generate(env=env)
```
Comment (franramirez688) [inline recipes/libselinux/all/conanfile.py:92]: Should we apply this only for cross-compilations?

**Review by franramirez688 — COMMENTED** [review]
Hi @jwillikers Thanks a lot for this PR (and the associated issue 😁 )

Sorry for the huge delay! I'm just revisiting all my old assigned issues! 🙏 

LGTM! I only have a question regarding to the environment.

File: recipes/libselinux/all/conanfile.py (hunk 1 above)
Comment (jwillikers) [reply recipes/libselinux/all/conanfile.py]: I don't see why not. I've updated this.

File: recipes/libselinux/all/conanfile.py [hunk 2]
```diff
@@ -87,7 +88,10 @@ def generate(self):
         sepol_lib_folder = os.path.join(self._sepol_source_folder, "src")
         tc.extra_ldflags.append(f"-L{sepol_lib_folder}")
         tc.make_args.append("USE_PCRE2=y")
-        tc.generate()
+        if cross_building(self):
+            env = tc.environment()
+            env.append_path("PKG_CONFIG_LIBDIR", self.generators_folder)
+            tc.generate(env=env)
```
Comment (franramirez688) [inline recipes/libselinux/all/conanfile.py:94]:
```suggestion
        env = tc.environment()
        if cross_building(self):
            env.append_path("PKG_CONFIG_LIBDIR", self.generators_folder)
        tc.generate(env=env)
```

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**